mantidproject/mantid
# Generalization of SampleLogsBehaviour (MergeRuns, ConjoinXRuns)

**Description of work.**

Currently, parameters of the parameter file specified by inflexible naming convention in `SampleLogsBehaviour.cpp` are used for `MergeRuns` and `ConjoinXRuns`. This may cause conflicts, if merging and joining is desired. This can be the case when running a reduction work-flow. For instance, `MergeRuns` parameters were specified and `Stitch1D` (uses `ConjoinXRuns` for point data) skipped the second to stitch workspace since the collimation angle was not identical in case of FIGARO data as shown in the issue. Thus, the behavior of merging runs and stitching or joining work-spaces does not agree.
This PR generalizes the algorithm `SampleLogsBehaviour` and adds new tests.

Default parameter names (formerly only parameter names) are used for `MergeRuns`
Prefix `conjoin_` + default parameter names are used for `ConjoinXRuns`
New algorithm can make use of new parameter names.

The user must modify the parameters file for changing and controlling the stitch behaviour since override properties are not exposed.

Please note code changes with regard to using structs containing strings and using shared MatrixWorkspace pointers in public methods. 

**To test:**

Tests are in place. Tests of `MergeRuns`, `ConjoinXRuns` did not change from a functional point of view.
Make sure the new `SampleLogsBehaviourTest` tests the new behaviour.

Fixes #23947.


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
